awx/conf/settings.py

```diff
@@ -255,6 +257,16 @@ def __init__(self, default_settings, cache, registry):
         self.__dict__['cache'] = EncryptedCacheProxy(cache, registry)
         self.__dict__['registry'] = registry
 
+    def __clean_on_fork__(self):
+        pid = os.getpid()
+        if pid not in self.__dict__['__forks__']:
+            self.__dict__['__forks__'][pid] = True
```

Comment:
We have to access this via `__dict__` instead of `getattr` to avoid a recursive loop (because `__getattr__` calls *this* function).

Reply:
I also tested this out by printing this dict in a tight loop of `awx-python` forks, and the data structure for each process never grows beyond two keys (the parent pid, and the current process pid).

So in effect, this code gets called *one time* when a child process does `getattr(settings, ...`), and then after that, it's a no-op.